### ISSUE TITLE
dsl: implement unit testing for DSL to WDL conversion of rooms, actions, and variables 

### ISSUE THREAD

**Issue by wzeng0** [uchicago-cs/chiventure#1255]
Right now, most testing is done simply by providing a DSL file and eyeballing the produced WDL file to check for discrepancies. It would be beneficial to have a more robust method to test the DSL to WDL parsing and conversion, so we hope to implement unit tests for certain aspects of a Chiventure game, including rooms, actions, and variables.

team members: Andi, William, Alex

**Comment by awliu2** [comment]
Not much progress has been made on this task. We will have to discuss whether to move forward with this task or not.

**Comment by awliu2** [comment]
Currently being worked on in the `dsl/unit_test` branch. Our testing framework has mostly been setup, now we are going to make it more user friendly, by implementing a default batch test, and also manually running specific tests from the command line.

**Comment by awliu2** [comment]
pull request created: https://github.com/uchicago-cs/chiventure/pull/1654

**Comment by alexlee0422** [comment]
In this task, we implement unit tests that tests the DSL to WDL parsing and conversion. This will be useful moving on since the only method to check if the conversions are right is by comparing manually. We added a tests/ subdirectory with a test.py testing file, as well as multiple dsl files and their expected outputs. We also added some flags that allow batch testing the dsl test files, individually testing them, and the --show flag which shows the difference between expected and output if there are any. (issue will be closed with the pull request)
